magento/pwa-studio
# Remove product from Wishlist

## Description

This PR adds the ability to remove products from wishlists.

Because the GraphQL support for wishlists doesn't arrive until Magento 2.4, the `removeProductsFromWishlist ` mutation doesn't actually modify the wishlists mock data the page is using currently.

It does still fire, and the hope is that once the mutation comes online there will be a minimal amount of effort to get it working properly.

## Related Issue





Closes PWA-630.

## Acceptance 



### Verification Stakeholders


@tjwiebell @schensley @dpatil-magento 

### Specification



### Verification Steps


1. Sign In
1. Navigate to the `/wishlist` page
1. Click the "More Actions" dots next to the "Add to Cart" call to action
1. Verify that a Dialog appears with "Move to", "Copy to", and "Remove" buttons
1. Click "Remove" 
1. Verify that you are presented with a confirmation prompt
1. Click "Cancel" and verify that no graphQL call appears in the `Network` tab
1. Try again but click "Delete" this time
1. Verify that a graphQL call appears in the `Network` tab (but fails because the page is using fake wishlist ids)
1. Verify the dialog buttons are disabled while the network request (and its retries) are in flight
1. Verify that an error message is shown after final `Network` tab failure
1. Verify that you can close the Dialog
1. Verify that the previous error message is not shown when you navigate back to the remove confirmation dialog

## Screenshots / Screen Captures (if appropriate)

<img width="242" alt="Screen Shot 2020-10-20 at 12 20 50 PM" src="https://user-images.githubusercontent.com/13182778/96622922-349af480-12d0-11eb-8cc3-1f1ea86c926f.png">

<img width="333" alt="Screen Shot 2020-10-20 at 12 21 24 PM" src="https://user-images.githubusercontent.com/13182778/96622939-38c71200-12d0-11eb-8acf-fa3fd359ddd3.png">

<img width="663" alt="Screen Shot 2020-10-20 at 12 20 58 PM" src="https://user-images.githubusercontent.com/13182778/96622938-38c71200-12d0-11eb-9b1f-9f2fb9614119.png">

## Checklist

* I have updated the documentation accordingly, if necessary.
- [x] Add confirmation prompt
- [x] Do not auto-close the Dialog on error
